### ISSUE TITLE
Any suggestions to finetune the mask-rcnn model to train own dataset? Which output layers should be tuned?

### ISSUE THREAD

**Issue by gabriellapizzuto** [multimodallearning/pytorch-mask-rcnn#11]

**Comment by krinj** [comment]
Yes, I was trying this today actually. You can load the pre-trained COCO model then fine-tune it, but you need to remove some layers from the state_dict.

You can exclude any layers that contain ["mask.conv5", "classifier.linear_class", "classifier.linear_bbox"] in its name. I personally also excluded all of the mask layers as well, but that's up to do.

Someone's pull request #12 already has this modification, or you can write your own.

**Comment by Oktai15** [comment]
@Infrarift could you briefly describe how to fine-tune this model on own dataset? It's not trivial for me

**Comment by austinmw** [comment]
Can someone please give me a hint on the fastest way to format my own data for training? My starting point is jpeg images along with binary-value .png image masks for each class, for each image.

**Comment by michhar** [comment]
Maybe this will help.  In my `XYZConfig(Config)` class I set 

```python
# Number of classes (including background)
NUM_CLASSES = 1 + 1
```

And to remove all keys that don’t match your model from the initialized state dict (with your config) and use it to load the weights afterwards:

```python
model = modellib.MaskRCNN(model_dir=MODEL_DIR, config=config)

pretrained_state = torch.load(COCO_MODEL_PATH)
model_state = model.state_dict()

pretrained_state = { k:v for k,v in pretrained_state.items() if k in model_state and v.size() == model_state[k].size() }
model_state.update(pretrained_state)
model.load_state_dict(model_state)
```

According to this post:  https://discuss.pytorch.org/t/how-to-load-part-of-pre-trained-model/1113/2.  Hope that's helpful.
